Mathlib/NumberTheory/IntervalSums.lean

```diff
@@ -0,0 +1,182 @@
+`Icc -N N`, `Ico -N N`, `Ioc -N N` respectively.
+We also prove that these filters are all `NeBot` and `LeAtTop`.
+
+-/
+
+open Finset Topology
+
+--Should this go elsewhere?
+lemma Finset.Icc_succ_succ (n : ℕ) : Icc (-(n + 1) : ℤ) (n + 1) = Icc (-n : ℤ) n ∪
+  {(-(n + 1) : ℤ), (n + 1 : ℤ)} := by
+  refine Finset.ext_iff.mpr ?_
+  intro a
+  simp only [neg_add_rev, Int.reduceNeg, mem_Icc, add_neg_le_iff_le_add, union_insert,
+    union_singleton, mem_insert]
+  omega
+
+lemma Finset.sum_Icc_of_even_eq_range {α : Type*} [CommRing α] {f : ℤ → α} (hf : ∀ n, f n = f (-n))
+    (N : ℕ) : ∑ m ∈ Icc (-N : ℤ) N, f m = 2 * ∑ m ∈ range (N + 1), f m - f 0 := by
+  induction N with
+  | zero => simp [two_mul]
+  | succ N ih =>
+    have := Icc_succ_succ N
+    simp only [neg_add_rev, Int.reduceNeg, Nat.cast_add, Nat.cast_one] at *
+    rw [this, sum_union (by simp), sum_pair (by omega), ih]
+    nth_rw 2 [sum_range_succ]
+    grind
+
+@[to_additive]
+lemma Finset.prod_Icc_eq_prod_Ico_succ {α : Type*} [CommMonoid α] (f : ℤ → α) {l u : ℤ}
+    (h : l ≤ u) : ∏ m ∈ Icc l u, f m = (∏ m ∈ Ico l u, f m) * f u := by
+  simp only [Icc_eq_cons_Ico h, cons_eq_insert, mem_Ico, lt_self_iff_false, and_false,
+    not_false_eq_true, prod_insert, mul_comm]
+
+lemma Finset.sum_Icc_add_endpoints {R : Type*} [AddCommGroup R] (f : ℤ → R) {N : ℕ} (hn : 1 ≤ N) :
+    ∑ m ∈ Icc (-N : ℤ) N, f m = f N + f (-N : ℤ) + ∑ m ∈ Icc (-(N - 1) : ℤ) (N - 1), f m := by
+  induction N
+  · grind
+  · zify
+    rw [Icc_succ_succ, sum_union (by simp)]
+    grind
+
+section IntervalFilters
+
+open Filter Function Finset SummationFilter
+
+variable (G : Type*) [Neg G] [Preorder G] [LocallyFiniteOrder G]
+
+/-- The SummationFilter on Locally finite order `G` corresponding to the symmetric
+intervals `Icc (-N) N`· -/
+def SummationFilter.symmetricConditional :
+    SummationFilter G where
+  filter := atTop.map (fun g ↦ Icc (-g) g)
+
+/-- The SummationFilter on `ℤ` corresponding to the intervals `Icc -N N`. Note that this is
+the same as the limit over open intervals `Ioo -N N` (see `SymmetricConditional_eq_map_Ioo`). -/
+abbrev SummationFilter.symCondInt : SummationFilter ℤ :=
+  SummationFilter.symmetricConditional ℤ
+
+lemma SymmetricConditional_eq_map_Icc :
+    symCondInt.filter = atTop.map (fun N ↦ Icc (-N) N) := rfl
+
+lemma SymmetricConditional_eq_map_Icc_nat :
+    symCondInt.filter = atTop.map (fun N : ℕ ↦ Icc (-(N : ℤ)) N) := by
+  rw [symCondInt, symmetricConditional, ← Nat.map_cast_int_atTop]
+  rfl
+
+/-- The SummationFilter on `G` corresponding to the intervals `Ico (-N) N`. -/
+def SummationFilter.IcoFilter : SummationFilter G where
+  filter := atTop.map (fun N ↦ Ico (-N) N)
+
+/-- The SummationFilter on `G` corresponding to the intervals `Ioc (-N) N`. -/
+def SummationFilter.IocFilter : SummationFilter G where
+  filter := atTop.map (fun N ↦ Ioc (-N) N)
+
+lemma SymmetricConditional_eq_map_Ioo :
+    (symmetricConditional ℤ).filter = atTop.map (fun N ↦ Ioo (-N) N) := by
```

Comment:
```suggestion
    symmCondInt.filter = atTop.map (fun N ↦ Ioo (-N) N) := by
```
(but I wonder if this is true more generally than `ℤ`)

Reply:
I asked Yael, and he seems to think this wont hold more generally. Mainly because apparently the only locally finite linearly ordered group is $$\mathbb{Z}$$! So this will have to do I guess.